keras/backend/theano_backend.py

```diff
@@ -324,8 +325,11 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, std, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, std, beta and gamma.
     '''
-    normed = T.nnet.bn.batch_normalization(x, gamma, beta, mean, std + epsilon,
-                                           mode='high_mem')
+    if theano.config.device.startswith('gpu'):
+        normed = dnn_batch_normalization_test(x, gamma, beta, mean, std ** 2, epsilon)
```

Comment:
It's irrelevant. Look at what the underlying implementation does with that
parameter and make sure that it matches the previous behavior. My guess is
that **2 should not be there.

On Jul 19, 2016 12:34, "Shashank Gupta" <notifications@github.com> wrote:

> In keras/backend/theano_backend.py
> <https://github.com/fchollet/keras/pull/3260#discussion_r71407795>:
>
> > @@ -324,8 +325,11 @@ def normalize_batch_in_training(x, gamma, beta,
> >  def batch_normalization(x, mean, std, beta, gamma, epsilon=0.0001):
> >      '''Apply batch normalization on x given mean, std, beta and gamma.
> >      '''
> > -    normed = T.nnet.bn.batch_normalization(x, gamma, beta, mean, std + epsilon,
> > -                                           mode='high_mem')
> > +    if theano.config.device.startswith('gpu'):
> > +        normed = dnn_batch_normalization_test(x, gamma, beta, mean, std ** 2, epsilon)
>
> This op takes variance as argument (
> http://deeplearning.net/software/theano_versions/dev/library/sandbox/cuda/dnn.html#batch-normalization)
> and this method takes std as argument, that's why **2. I didn't at mode
> properly. I should have actually.
>
> batch normalization implementation in theano (
> deeplearning.net/software/theano/library/tensor/nnet/bn.html) is
> per-activation, right?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/fchollet/keras/pull/3260/files/52599b1a35dc8144c17e665b06a91ad997d5f51e#r71407795>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AArWb9OzMmbYo464jtGvGJPQg1KP0flEks5qXSbSgaJpZM4JP8_B>
> .
>